### ISSUE TITLE
buffer-history-tree command fails

### ISSUE THREAD

**Issue by bubbleguuum** [atlas-engineer/nyxt#2357]
Using latest master as of the date of this report.

Command `buffer-history-tree` (mentioned in the tutorial) fails loading the page, giving that output in the console:

```
<WARN> [21:06:37] Expected buffer, got NIL
<WARN> [21:06:37] Warning: Error while processing the "nyxt:" URL: The value of STRING is NIL, which is not of type STRING.
<INFO> [21:06:37] Failed to load URL nyxt:nyxt/history-mode:buffer-history-tree?id=%1B%23%3A%7C890%7C in buffer 895.
```

**Comment by aartaka** [comment]
@Ambrevar, this (some other issues) seems to stem from symbol-based buffer ID, because `(gethash '#:|1150| (slot-value nyxt:*browser* 'nyxt:buffers))` returns `(values nil nil)` with the buffer ID 1150 being present in the table. Maybe use more conventional of a symvols, like plain `gensym` ones? Or keywords?

**Comment by aartaka** [comment]
See [the Hyperspec](http://www.lispworks.com/documentation/HyperSpec/Body/f_eq.htm)
```
Symbols that print the same usually are eq to each other because of the use of the intern function.
```

Which means that uninterned symbols are not necessarily `eq` to each other and are a poor choice for hash-table keys.

**Comment by aartaka** [comment]
Interestingly, changing `new-id` to 
``` lisp
(defun new-id ()
  (intern (symbol-name (gensym "id")) :keyword))
```
Fixes this issue and the one described by @aadcg in https://github.com/atlas-engineer/nyxt/pull/2333#issuecomment-1145214151.

It seems that hash-tables don't exactly work well with non-interned symbols. See
```
CL-USER> (defvar ht (make-hash-table))
HT
CL-USER> (setf (gethash '#:hello ht) 4)
4 (3 bits, #x4, #o4, #b100)
CL-USER> (gethash '#:hello ht)
NIL
NIL
```

**Comment by aartaka** [comment]
Ooooops, wrong comment order 😅

**Comment by Ambrevar** [comment]
Duh, I just saw your message after posting https://github.com/atlas-engineer/nyxt/issues/2361 :p

**Comment by Ambrevar** [comment]
Fixed with 37a61ccca3c6bd0eb4e952212a747093bad48ee8.
